### PR TITLE
Support filtering list items by name in MultiLevelSelector

### PR DESCRIPTION
**Pull request by samschaevitz**
## Description

Add a search bar and ability to filter MultiLevelSelector based on item names. 

## Todo

- [ ] PR has a meaningful title?
- [ ] Lint passes?
- [ ] Ran `yarn format` in frontend?

### CODE REVIEW

File: frontend/pages/faq.js [hunk 1]
```diff
@@ -90,7 +90,9 @@ export default function Faq({ questionsBySection, questions }) {
   // The first section should be the initial tab value
   const [searchValue, setSearchValue] = React.useState("");
 
-  const handleSearchBarChange = (type, value) => setSearchValue(value);
+  const handleSearchBarChange = (event) => {
+    setSearchValue(event.target.value);
```
Comment (piperchester) [inline frontend/pages/faq.js:94]: I _believe_ this won't be necessary in most cases, but WDYT_ about being oversafe around `ReferenceError` and using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here? Chris and I have been nerding out about it being our newest favorite JS feature 😄: 
```suggestion
    setSearchValue(event?.target?.value);
```

File: frontend/src/components/general/MultiLevelSelector.js [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
-
-import { List, ListItem, ListItemText, ListItemIcon, Typography, Divider } from "@material-ui/core";
+import {
+  Container,
+  Divider,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Typography,
```
Comment (piperchester) [inline frontend/src/components/general/MultiLevelSelector.js:8]: Nice! Not sure if it was intentional but I like the alphabetization. Helps with readability and thereby understandability IMO 🙏

**Review by piperchester — COMMENTED** [review]
This is looking great! Excited about this feature. Not sure if it it's still in draft / WIP or ready for a full review so just leaving a couple flyby comments.

File: frontend/pages/faq.js (hunk 1 above)
Comment (samschaevitz) [reply frontend/pages/faq.js:94]: SGTM :)

File: frontend/src/components/general/MultiLevelSelector.js (hunk 2 above)
Comment (samschaevitz) [reply frontend/src/components/general/MultiLevelSelector.js:8]: This is thanks to my Prettier extension in VSCode. Glad you like it! Eventually would we want something like this in our contributor style guide?

**Comment by samschaevitz** [comment]
This is working as far as I can tell. Is there somewhere I should add some tests? Or is that a larger change (thinking about our discussion at the last dev meeting)?

File: frontend/src/components/general/MultiLevelSelector.js (hunk 2 above)
Comment (samschaevitz) [reply frontend/src/components/general/MultiLevelSelector.js:8]: JK, it's actually just VSCode editor settings

```
 "editor.codeActionsOnSave": {
        "source.fixAll": true,
        "source.organizeImports": true,
        "source.addMissingImports": true,
    },
```

**Review by positiveimpact — APPROVED** [review]
LGTM! Great job, feel free to merge :)
I wouldn't add frontend test at the moment. If I remember correctly we just wanted to do backend unit tests in the short term and eventually e2e testing.

**Comment by ddhanesha** [comment]
> LGTM! Great job, feel free to merge :)
> I wouldn't add frontend test at the moment. If I remember correctly we just wanted to do backend unit tests in the short term and eventually e2e testing.

Yes, end to end testing makes more sense to me than frontend unittests. But @samschaevitz if you have pros of writing frontend unittests I would happy to learn about that more.

**Pull request merged**